### PR TITLE
fix: Unable to modify the location of a booking when rescheduling

### LINKED ISSUE #11650: Unable to modify the location of a booking when rescheduling
### Issue Summary

Having set a event's location to be "In Person (Attendee Address)", trying to modify it during a reschedule does not work. I noticed the input field for the location is set to `readonly` but not `disabled`.

### Steps to Reproduce

1. Create a new event type
2. Set Location to be "In Person (Attendee Address)"
3. Save the event 
4. View public page and open the event type
5. Schedule a booking with the address filled in
6. Reschedule
7. The location cannot be modified

I consider this a bug because the location of a meeting is bound to be updated.

### Actual Results

- Location input field cannot be modified and is set to read-only
- Cursor over input field still shows the test selection (i.e. `cursor: text`)
- _Interesting_, removing the readonly and modifying it actually causes location to be committed to the backend, albeit not entirely.
  - The location under "Where" still shows the value pre-modification
  - However, when rescheduling, the location input actually shows the value post-modification

image

### Expected Results

The location should be modifiable and reflected in the booking confirmation and associated webhooks.

### Technical details

- Chrome Version 117.0.5938.92

### Evidence
[cal-reschedule-location-bug.webm](https://github.com/calcom/cal.com/assets/577802/8996a4d9-7902-4875-b4f9-8b133443508f)

### PR DESCRIPTION
**Pull request by kremedev**
/claim #11650

## What does this PR do?

Fixes #11650

https://github.com/calcom/cal.com/assets/134702704/a201baae-61cd-48ac-acd1-aa9c9254e0c6

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: apps/web/pages/booking/[uid].tsx [hunk 1]
```diff
@@ -115,6 +115,13 @@ export default function Success(props: SuccessProps) {
   const tz = props.tz ? props.tz : isSuccessBookingPage && attendeeTimeZone ? attendeeTimeZone : timeZone();
 
   const location = props.bookingInfo.location as ReturnType<typeof getEventLocationValue>;
+  let rescheduleLocation: string | undefined;
+  if (
+    typeof props.bookingInfo.responses.location === "object" &&
+    "optionValue" in props.bookingInfo.responses.location
+  ) {
+    rescheduleLocation = props.bookingInfo.responses.location.optionValue;
+  }
```
Comment (kremedev) [inline apps/web/pages/booking/[uid].tsx:124]: How about this?
```
  const rescheduleLocation: string =
    typeof props.bookingInfo.responses.location === "object" &&
    "optionValue" in props.bookingInfo.responses.location
      ? props.bookingInfo.responses.location.optionValue
      : "";```

**Review by CarinaWolli — APPROVED** [review]
Looks good! Thank you for your contribution 🙏

**Comment by hariombalhara** [comment]
@kremedev Does it send an email informing about location change as well? I kept it disabled earlier because that flow wasn't there at that time.

**Comment by kremedev** [comment]
Sorry for the late reply, I was at school.

> @kremedev Does it send an email informing about location change as well? I kept it disabled earlier because that flow wasn't there at that time.

No, I wasn't informed about that. I was waiting for a review.

**Pull request merged**